activerecord/test/cases/adapters/postgresql/optimizer_hints_test.rb

```diff
@@ -19,6 +19,14 @@ def test_optimizer_hints
       end
     end
 
+    def test_optimizer_hints_with_count_subquery
+      assert_sql(%r{\ASELECT /\*\+ SeqScan\(posts\) \*/}) do
+        posts = Post.optimizer_hints("SeqScan(posts)")
+        posts = posts.select(:id).where(author_id: [0, 1]).limit(5)
+        assert_equal 5, posts.count
+      end
+    end
```

Comment:
Similar issue for PostgreSQL: https://pg-hint-plan.readthedocs.io/en/latest/hint_details.html#subqueries